### PR TITLE
4851 reject labmessage

### PR DESCRIPTION
**Pull request by leventegal-she**
Closes https://github.com/hzi-braunschweig/SORMAS-Project/issues/4851

### CODE REVIEW

File: sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageForm.java [hunk 1]
```diff
@@ -10,6 +10,8 @@
 import de.symeda.sormas.api.FacadeProvider;
 import de.symeda.sormas.api.labmessage.ExternalMessageResult;
 import de.symeda.sormas.api.labmessage.LabMessageDto;
+import de.symeda.sormas.api.labmessage.LabMessageStatus;
```
Comment (Alexandru-Pirvu) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageForm.java:13]: Not used

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 2]
```diff
@@ -7124,3 +7124,18 @@ ALTER TABLE contact_history ADD CONSTRAINT fk_contact_followupstatuschangeuser_i
 INSERT INTO schema_version (version_number, comment) VALUES (358, '2021-04-06 Add date and responsible user of last follow-up status change #4138');
 
 -- *** Insert new sql commands BEFORE this line ***
```
Comment (Alexandru-Pirvu) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:7126]: This should be deleted

File: sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessagesGridComponent.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.labmessage.LabMessageCriteria;
+import de.symeda.sormas.api.labmessage.LabMessageStatus;
+import de.symeda.sormas.ui.ViewModelProviders;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.LayoutUtil;
+
+public class LabMessagesGridComponent extends VerticalLayout {
```
Comment (Alexandru-Pirvu) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessagesGridComponent.java:24]: Is this used?

**Review by Alexandru-Pirvu — CHANGES_REQUESTED** [review]

File: sormas-api/src/main/resources/captions.properties [hunk 4]
```diff
@@ -141,6 +141,8 @@ actionSettings=User Settings
 actionNewForm=New Form
 actionOverwrite=Overwrite
 actionRemindMeLater=Remind me later
+actionUnclearLabMessage=Mark as unclear
+actionManualForwardLabMessage=Mark as manually forwarded
```
Comment (FredrikSchaeferVitagroup) [inline sormas-api/src/main/resources/captions.properties:145]: I find this slightly confusing. Actually, the lab message is just marked as forwarded, not as MANUALLY forwarded. The status makes no distinction between forwarded via S2S and manually forwarded. I would therefore suggest to change this to 
```suggestion
actionManualForwardLabMessage=Mark as forwarded
```

File: sormas-api/src/main/resources/strings.properties [hunk 5]
```diff
@@ -192,6 +192,8 @@ confirmationLocationFacilityAddressOverride = The selected facility has location
 confirmationCancelExternalFollowUpPopup=Are you sure you want to cancel external follow-ups in the eDiary?
 confirmationUnlinkCaseFromEvent = Are you sure you want to unlink this case from this event?
 confirmationDeleteLabMessages = Are you sure you want to delete all %d selected lab messages?
+confirmationUnclearLabMessage = Are you sure you want to mark this lab message as unclear?
+confirmationManuallyForwardedLabMessage = Are you sure you want to mark this lab message as manually forwarded?
```
Comment (FredrikSchaeferVitagroup) [inline sormas-api/src/main/resources/strings.properties:196]:
```suggestion
confirmationManuallyForwardedLabMessage = Are you sure you want to mark this lab message as forwarded?
```

**Review by FredrikSchaeferVitagroup — CHANGES_REQUESTED** [review]

**Review by FredrikSchaeferVitagroup — APPROVED** [review]

**Pull request merged**